vm/src/stdlib/os.rs

```diff
@@ -1127,6 +1130,123 @@ pub(super) mod _os {
         OutputMode::String.process_path(curdir_inner(vm)?, vm)
     }
 
+    #[cfg(unix)]
+    #[derive(FromArgs)]
+    struct RegisterAtForkArgs {
+        #[pyarg(named, optional)]
+        before: OptionalArg<PyObjectRef>,
+        #[pyarg(named, optional)]
+        after_in_parent: OptionalArg<PyObjectRef>,
+        #[pyarg(named, optional)]
+        after_in_child: OptionalArg<PyObjectRef>,
+    }
```

Comment:
I don't have a clever idea either. But this is my suggestion to isolate validation and logic.
```suggestion
    }
    
    impl RegisterAtForkArgs {
        fn into_validated(self, vm: &VirtualMachine) -> PyResult<(Option<PyObjectRef>, Option<PyObjectRef>, Option<PyObjectRef>)> {
            fn into_option(arg: OptionalArg<PyObjectRef>, vm: &VirtualMachine) -> PyResult<Option<PyObjectRef>> {
                match arg {
                    OptionalArg::Present(obj) => {
                        if !obj.is_callable() {
                            return Err(vm.new_type_error("Args must be callable".to_owned()));
                        }
                        Ok(Some(obj))
                    }
                    OptionalArg::Missing => Ok(None),
                }
            }
            let before = into_option(self.before, vm)?;
            let after_in_parent = into_option(self.after_in_parent, vm)?;
            let after_in_child = into_option(self.after_in_child, vm)?;
            if before.is_none() && after_in_parent.is_none() && after_in_child.is_none() {
                return Err(vm.new_type_error("At least one arg must be present".to_owned()));
            }
            Ok((before, after_in_parent, after_in_child))
        }
    }
```